File: HackerRank Solutions/Interview Preparation Kit/Miscellaneous/Time Complexity Primality.py
#!/bin/python3

import math
import os
import random
import re
import sys
from math import sqrt
# Complete the primality function below.
def primality(n):
    if(not n%2 and n!=2 or n==1):
        return "Not prime"
    for i in range(3, int(sqrt(n))+1, 2):
        if(not n%i):
            return "Not prime"
    return "Prime"
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    p = int(input())

    for p_itr in range(p):
        n = int(input())

        result = primality(n)

        fptr.write(result + '\n')

    fptr.close()

File: HackerRank Solutions/Interview Preparation Kit/Miscellaneous/Time Complexity Primality.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the primality function below.
string primality(int n) {
    if(n%2==0 && n!=2 || n==1){
        return "Not prime";
    }
    for(int i=2; i<(int)sqrt(n)+1; ++i){
        if(n%i==0){
            return "Not prime";
        }
    }
    return "Prime";
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int p;
    cin >> p;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int p_itr = 0; p_itr < p; p_itr++) {
        int n;
        cin >> n;
        cin.ignore(numeric_limits<streamsize>::max(), '\n');

        string result = primality(n);

        fout << result << "\n";
    }

    fout.close();

    return 0;
}
